File: app/src/main/java/com/simcom/ecashier/ui/history/HistoryViewModel.java
package com.simcom.ecashier.ui.history;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.simcom.ecashier.model.repositories.Repository;

public class HistoryViewModel extends AndroidViewModel {


    public HistoryViewModel(Application application) {
        super(application);
        Repository repository = new Repository(application);
    }


}

File: app/src/main/java/com/simcom/ecashier/ui/history/HistoryViewModel.kt
package com.simcom.ecashier.ui.history

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.room.Room
import com.simcom.ecashier.model.room.CashierDatabase
import com.simcom.ecashier.model.room.Collection

class HistoryViewModel(application: Application) : AndroidViewModel(application) {
    private var db: CashierDatabase = Room.databaseBuilder(
        application,
        CashierDatabase::class.java, "cashier_database"
    ).fallbackToDestructiveMigration().build()

    fun getAllCollections() : LiveData<List<Collection>> {
        return db.collectionDao().getAllCollections()
    }
}